src/test/java/api/requests/scenarios/RequestPolicyTests.java

```diff
@@ -0,0 +1,378 @@
+
+    //an undergraduate student attempts to place a Recall request
+    final IndividualResource recallRequest = requestsClient.create(new RequestBuilder()
+      .recall()
+      .forItem(checkedOutItem)
+      .withPickupServicePointId(requestPickupServicePoint.getId())
+      .by(usersFixture.undergradHenry()));
+
+    JsonObject requestedItem = recallRequest.getJson().getJsonObject("item");
+    assertThat(recallRequest.getJson().getString("requestType"), is(RequestType.RECALL.getValue()));
+    assertThat(requestedItem.getString("status"), is(ItemStatus.CHECKED_OUT.getValue()));
+    assertThat(recallRequest.getJson().getString("status"), is(RequestStatus.OPEN_NOT_YET_FILLED.getValue()));
+  }
+
+  @Test
+  public void cannotCreateRecallRequestsWithRequestPolicyNotAllowingRecalls()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    final String undergradPatronGroupPolicy = patronGroupsFixture.undergrad().getId().toString();
+    final String anyNoticePolicy = noticePoliciesFixture.activeNotice().getId().toString();
+    final String anyLoanPolicy = loanPoliciesFixture.canCirculateRolling().getId().toString();
+    final String anyRequestPolicy = requestPoliciesFixture.allowAllRequestPolicy().getId().toString();
+
+    ArrayList<RequestType> allowedRequestTypes = new ArrayList<>();
+    allowedRequestTypes.add(RequestType.HOLD);
+    allowedRequestTypes.add(RequestType.PAGE);
+    final String noRecallRequestPolicy = requestPoliciesFixture.customRequestPolicy(allowedRequestTypes,
+                                          "All But Recall", "All but Recall request policy").getId().toString();
+
+    //This rule is set up to show that the fallback policy won't be used but the patronGroup policy g is used instead.
+    //The patronGroup policy allows undergraduate students to place a request on any material type, loan or notice type, on any request type
+    //except for RECALL
+    final String rules = String.join("\n",
+      "priority: t, s, c, b, a, m, g",
+      "fallback-policy : l " + anyLoanPolicy + " r " + anyRequestPolicy + " n " + anyNoticePolicy + "\n",
+      "g " + undergradPatronGroupPolicy + ": l " + anyLoanPolicy + " r " + noRecallRequestPolicy +" n " + anyNoticePolicy
+    );
+
+    setRules(rules);
+
+    //setting up a checked-out library item to perform recall
+    final IndividualResource checkedOutItem = itemsFixture.basedUponSmallAngryPlanet();
+    loansFixture.checkOut(checkedOutItem, usersFixture.jessica());
+
+    final Response recallResponse = requestsClient.attemptCreate(new RequestBuilder()
+      .recall()
+      .forItem(checkedOutItem)
+      .withPickupServicePointId(requestPickupServicePoint.getId())
+      .by(usersFixture.undergradHenry()));
+
+    assertThat(recallResponse, hasStatus(HTTP_VALIDATION_ERROR));
+    assertThat(recallResponse.getJson(), hasErrorWith(allOf(
+      hasMessage("Request Type Recall is not valid"))));
```

Comment:
This does not fit the structure of the message expected in UIREQ-211 - `Recall requests are not allowed for this patron and item combination`

I believe this needs to be as it is what is going to be presented to the user directly by the UI.

Please can this be changed. 

Please can the error messages for the whitelist also be changed to match this format as well?

Reply:
@marcjohnson-kint By changing the whitelist error messages you mean the message returned by CreateRequestService.refuseWhenItemIsNotValid()? I don't mind changing it but I wonder if this will affect the UI implementation as this message must have already been used by now. If the UI only displayed it then we're good, if it makes decision on it then we might need to reconsider.

Reply:
Yes, that is the method I meant.

The UI does not use it at all anymore, except in unexpected situations (where it should just display it), as it implemented the whitelist itself.

However we agreed to carry over (so we didn't revisit the previous work once this was started) changing the backend messages to match those for the request policy into this work, to make it consistent.

If this isn't viable we can defer it to a later change?

Reply:
Updated the error message texts in both refuseWhenItemIsNotValid() and refuseWhenRequestCannotBeFulfilled()